python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py

```diff
@@ -1,16 +1,124 @@
 import os
+import re
 from itertools import chain
 from typing import Any, Callable, Dict, List, Mapping, NamedTuple, Optional, Set, cast
 
 import yaml
+from dagster_airbyte.resources import AirbyteResource
 from dagster_airbyte.utils import generate_materializations
 
-from dagster import AssetKey, AssetOut, Output
+from dagster import AssetKey, AssetOut, Output, ResourceDefinition
 from dagster import _check as check
+from dagster import with_resources
 from dagster._annotations import experimental
 from dagster._core.definitions import AssetsDefinition, multi_asset
+from dagster._core.definitions.assets_lazy import AssetsDefinitionMetadata, LazyAssetsDefinition
 from dagster._core.definitions.events import CoercibleToAssetKeyPrefix
 from dagster._core.definitions.load_assets_from_modules import with_group
+from dagster._core.execution.context.init import build_init_resource_context
+
+
+def _build_airbyte_asset_defn_metadata(
+    connection_id: str,
+    destination_tables: List[str],
+    asset_key_prefix: Optional[List[str]] = None,
+    normalization_tables: Optional[Mapping[str, List[str]]] = None,
+    upstream_assets: Optional[Set[AssetKey]] = None,
+    group_name: Optional[str] = None,
+) -> AssetsDefinitionMetadata:
+
+    asset_key_prefix = check.opt_list_param(asset_key_prefix, "asset_key_prefix", of_type=str)
+
+    # Generate a list of outputs, the set of destination tables plus any affiliated
+    # normalization tables
+    tables = chain.from_iterable(
+        chain([destination_tables], normalization_tables.values() if normalization_tables else [])
+    )
+    outputs = {table: AssetKey(asset_key_prefix + [table]) for table in tables}
+
+    internal_deps = {}
+
+    normalization_tables = (
+        {k: list(v) for k, v in normalization_tables.items()} if normalization_tables else {}
+    )
+
+    # If normalization tables are specified, we need to add a dependency from the destination table
+    # to the affilitated normalization table
+    if normalization_tables:
+        for base_table, derived_tables in normalization_tables.items():
+            for derived_table in derived_tables:
+                internal_deps[derived_table] = {AssetKey(asset_key_prefix + [base_table])}
+
+    # All non-normalization tables depend on any user-provided upstream assets
+    for table in destination_tables:
+        internal_deps[table] = upstream_assets or set()
+
+    return AssetsDefinitionMetadata(
+        input_keys=set(),
+        output_keys=outputs,
+        asset_deps=internal_deps,
+        group_names_by_key={table: group_name for table in tables} if group_name else None,
+        metadata={
+            "connection_id": connection_id,
+            "group_name": group_name,
```

Comment:
this feels a little bad to pass, since `multi_assets` just takes a single group name